src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/resource/AddElementDialog.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from "react";
+import CloseIcon from "@mui/icons-material/Close";
+import { Button } from "@madie/madie-design-system/dist/react";
+import { Dialog, IconButton, DialogActions, Divider } from "@mui/material";
+import ElementSelector from "../element/ElementSelector";
+import { ElementDefinition } from "fhir/r4";
+
+export interface AddElementDialogProps {
+  open: boolean;
+  onClose: any;
+  basePath: string;
+  options: ElementDefinition[];
+  value: ElementDefinition[];
+  saveElements: Function;
+}
+
+const AddElementDialog = (props: AddElementDialogProps) => {
+  const { open, onClose, basePath, options, value, saveElements } = props;
+
+  const [newValues, setNewValues] = useState<ElementDefinition[]>([]);
+  useEffect(() => {
+    setNewValues(value);
+  }, [value]);
+  const handleChange = (event, newValue: ElementDefinition[] | null) => {
```

Comment:
handleChange, handleClose and handleSave are recreated on each render. we can optimize them using useCallback to prevent unnecessary re-renders of the child components.